app/src/main/java/nl/tudelft/cs4160/trustchain_android/network/MessageHandler.java

```diff
@@ -102,7 +102,7 @@ public void handlePuncture(Peer peer, MessageProto.Puncture puncture) throws IOE
      */
     public void handleReceivedBlock(Peer peer, MessageProto.TrustChainBlock block) {
         try {
-            if (TrustChainBlockHelper.validate(block,dbHelper).getStatus() != ValidationResult.INVALID ) {
+            if (dbHelper != null && TrustChainBlockHelper.validate(block,dbHelper).getStatus() != ValidationResult.INVALID ) {
                 dbHelper.replaceInDB(block);
```

Comment:
Maybe add some debug log here that states that the dbHelper is null for easier debugging.

Reply:
done